### PR TITLE
More modules

### PR DESCRIPTION
**Pull request by 12944qwerty**
Currently includes:
 
- [x] Category Component
- [x] FormItem, FormText, and Divider Components (all for settings)
- [x] SwitchItem Component
- [x] ContextMenu Module
- [x] ContextMenu Components (not all of the components are included yet)
- [x] Modal Components and Module
- [x] Type FluxDispatcher
- [x] Tooltip Component
- [x] Button and ButtonItem Components

### CODE REVIEW

**Review by SammCheese — APPROVED** [review]
Looks good, submittive and readable!

File: src/main/index.ts
Comment (asportnoy) [inline src/main/index.ts:151]: Let's just delete this

File: src/renderer/modules/components/Category.tsx
Comment (asportnoy) [inline src/renderer/modules/components/Category.tsx:28]: Self explanitory
```suggestion
```

File: src/renderer/modules/components/Category.tsx
Comment (asportnoy) [inline src/renderer/modules/components/Category.tsx:57]: Merge master in to this branch, this may not be needed anymore.

File: src/renderer/modules/components/Category.tsx
Comment (asportnoy) [inline src/renderer/modules/components/Category.tsx:7]: Decrease to `5_000` for consistency with common modules, assuming that doesn't cause issues. Same for all other timeouts.

File: src/renderer/modules/components/Modal.tsx
Comment (asportnoy) [inline src/renderer/modules/components/Modal.tsx:11]: Can we make better types for this

File: src/renderer/modules/webpack/common/modal.ts
Comment (asportnoy) [inline src/renderer/modules/webpack/common/modal.ts:1]: What's the difference between this and the modal component? Shouldn't those just be combined?

File: src/renderer/modules/webpack/index.ts
Comment (asportnoy) [inline src/renderer/modules/webpack/index.ts:27]: Bad merge?

**Review by asportnoy — DISMISSED** [review]

File: src/renderer/modules/webpack/common/modal.ts
Comment (12944qwerty) [reply src/renderer/modules/webpack/common/modal.ts:1]: This has the open, close, etc. These make the modals appear on the screen. The components is what is shown. They're different

File: src/renderer/modules/webpack/index.ts
Comment (FifiTheBulldog) [inline src/renderer/modules/webpack/index.ts:791]: See above.
```suggestion
```

File: src/renderer/modules/webpack/index.ts
Comment (FifiTheBulldog) [inline src/renderer/modules/webpack/index.ts:51]: This entire namespace is completely unnecessary and just adds bloat to our API and our code.
```suggestion
```

File: src/renderer/modules/webpack/index.ts
Comment (FifiTheBulldog) [inline src/renderer/modules/webpack/index.ts:89]: Why does this need to be exported? It's meant to be private, for our own functions to be able to find modules by string.

File: src/renderer/modules/webpack/index.ts
Comment (FifiTheBulldog) [inline src/renderer/modules/webpack/index.ts:164]: See above. No webpack namespace.
```suggestion
```

**Review by FifiTheBulldog — CHANGES_REQUESTED** [review]
Some good stuff, but please remove the bloat from webpack. That module is complicated enough already; we will not add unnecessary code to it.

File: src/renderer/modules/components/index.ts
Comment (FifiTheBulldog) [inline src/renderer/modules/components/index.ts:16]: Does this need to be 10 seconds? That's a long time.

**Review by FifiTheBulldog — APPROVED** [review]
Looks pretty good to me. I have just one question, addressed in a separate comment. I'll let you merge this when you're ready.

File: src/renderer/modules/components/index.ts
Comment (12944qwerty) [reply src/renderer/modules/components/index.ts:16]: 5 seconds wasn't very consistent so i did 10

Comment (FifiTheBulldog) [reply src/renderer/modules/components/index.ts:16]: Fair enough

File: src/renderer/modules/components/Category.tsx
Comment (asportnoy) [inline src/renderer/modules/components/Category.tsx:19]: > It's, by default, is automatically handled by the component.

File: src/renderer/modules/webpack/common/index.ts
Comment (asportnoy) [inline src/renderer/modules/webpack/common/index.ts:29]: Doesn't work properly in the docs https://docs.replugged.dev/variables/common.html

File: src/renderer/modules/components/index.ts
Comment (asportnoy) [inline src/renderer/modules/components/index.ts:1]: Doesn't work in the docs https://docs.replugged.dev/variables/components.html

File: src/renderer/modules/components/Modal.tsx
Comment (asportnoy) [reply src/renderer/modules/components/Modal.tsx:60]: Not resolved yet

File: src/renderer/modules/webpack/index.ts
Comment (asportnoy) [reply src/renderer/modules/webpack/index.ts:56]: Not yet resolved

File: src/renderer/modules/webpack/common/index.ts
Comment (lexisother) [reply src/renderer/modules/webpack/common/index.ts:29]: (pretty sure this can be resolved with `export * as constants from "./constants"`)

**Pull request merged**
